python/desc/imsim/imSim_class_factory.py

```diff
@@ -52,8 +57,16 @@ def imSim__init__(self, phosim_objects, obs_metadata, catalog_db=None):
     self.db_obj = type('DummyDB', (), dict(epoch=2000))
 
     # Add noise and sky background
-    self.noise_and_background = ExampleCCDNoise(addNoise=True,
-                                                addBackground=True)
+    # The simple code using the default lsst-GalSim interface would be:
+    #
+    #    self.noise_and_background = ExampleCCDNoise(addNoise=True,
+    #                                                addBackground=True)
+    #
+    # But, we need a more realistic sky model and we need to pass more than
+    # this basic info to use Peter Y's ESO sky model.
+    # We must pass obs_metadata, chip information etc...
+    self.noise_and_background = ESOSkyModel(obs_metadata, addNoise=True,
```

Comment:
Creating images with different object types is always going to be awkward because of the way objects are stored on fatboy (different tables for stars, galaxies, asteroids, etc.).

`sims_GalSimInterface` includes a method `copyGalSimInterpreter()` in `GalSimCatalog` which allows you to copy the `self.galSimInterpreter` (which actually does the image generation) from catalog to another to ensure that all of the objects land in the same FITS files.  We could amend that method so that it also copies `self.PSF` and `self.noise_and_background` from one `GalSimCatalog` to another.